test/e2e/utils/recordedit-helpers.js

```diff
@@ -321,7 +347,51 @@ exports.testPresentationAndBasicValidation = function(tableParams, isEditMode) {
                     });
                 });
             }
+            
+            if (jsonCols.length > 0) {
+                describe("JSON fields, ", function () {
+                    it("should show textarea input for JSON datatype and then set the value", function() {
+                        jsonCols.forEach(function(c) {
+                        chaisePage.recordEditPage.getTextAreaForAcolumn(c.name, recordIndex).then(function(jsonTxtArea) {
+                            if (jsonTxtArea) {
+                                expect(true).toBeDefined();
+                                jsonTxtArea.column = c;
+                                
+                                JSONDataTypeFields.push(jsonTxtArea);
+                                var value = getRecordValue(c.name);
 
+                                if (value != undefined) {
+                                    expect(jsonTxtArea.getAttribute('value')).toBe(value, colError(c.name , "Doesn't have the expected value."));
+                                }              
+                            }
+                            else{
+                                expect(undefined).toBeDefined();
+                            }
+                        });
+                    });
+                 });
+                 
+                 it("should only allow valid JSON values", function(){
+                     jsonCols.forEach(function(c) {
+                         chaisePage.recordEditPage.getTextAreaForAcolumn(c.name, recordIndex).then(function(jsonTxtArea) {
+                             for (i = 0; i < JSONTestParams.length; i++) {
+                                 jsonTxtArea.clear();
+                                 (function(input){
+                                     c._value = input;
+                                     jsonTxtArea.sendKeys(input);
+                                     chaisePage.recordEditPage.getJSONInputErrorMessage(jsonTxtArea, 'json').then(function(error){
+                                         if(error){
```

Comment:
@jrchudy This is also something that has been written before and @nitishksahu has used those as an example. We can remove the `if() {} else {}` the same way that you suggested on the other comment.